src/SensorReading.ts

```diff
@@ -58,43 +72,44 @@ export class SensorReading {
     }
     return 0;
   }
+
+  static pmToAQandU(pm: number): number {
+    return this.pmToAQI(0.778 * pm + 2.65);
+  }
+
+  static pmToLRAPA(pm: number): number {
+    return this.pmToAQI(0.5 * pm - 0.66);
+  }
   
   static pmToAQI(pm: number): number {
-    // This calculation was lifted from https://github.com/SANdood/homebridge-purpleair.
     let aqi: number;
-    if (pm > 500) {
-      aqi = 500;
-    } else if (pm > 350.5) {
-      aqi = this.remap(pm, 350.5, 500.5, 400, 500);
+
+    if (pm > 350.5) {
+      aqi = this.calcAQI(pm, 500, 401, 500, 350.5);
     } else if (pm > 250.5) {
-      aqi = this.remap(pm, 250.5, 350.5, 300, 400);
+      aqi = this.calcAQI(pm, 400, 301, 350.4, 250.5);
     } else if (pm > 150.5) {
-      aqi = this.remap(pm, 150.5, 250.5, 200, 300);
+      aqi = this.calcAQI(pm, 300, 201, 250.4, 150.5);
     } else if (pm > 55.5) {
-      aqi = this.remap(pm, 55.5, 150.5, 150, 200);
+      aqi = this.calcAQI(pm, 200, 151, 150.4, 55.5);
     } else if (pm > 35.5) {
-      aqi = this.remap(pm, 35.5, 55.5, 100, 150);
-    } else if (pm > 12) {
-      aqi = this.remap(pm, 12, 35.5, 50, 100);
-    } else if (pm > 0) {
-      aqi = this.remap(pm, 0, 12, 0, 50);
+      aqi = this.calcAQI(pm, 150, 101, 55.4, 35.5);
+    } else if (pm > 12.1) {
+      aqi = this.calcAQI(pm, 100, 51, 35.4, 12.1);
+    } else if (pm >= 0) {
+      aqi = this.calcAQI(pm, 50, 0, 12, 0);
     } else {
-      aqi = 0; 
+      aqi = 0;
     }
+
     return aqi;
   }
 
-  static remap(value: number, fromLow: number, fromHigh: number, toLow: number, toHigh: number): number {
-    // This calculation was lifted from https://github.com/SANdood/homebridge-purpleair.
-    const fromRange = fromHigh - fromLow;
-    const toRange = toHigh - toLow;
-    const scaleFactor = toRange / fromRange;
-
-    // Re-zero the value within the from range
-    let tmpValue = value - fromLow;
-    // Rescale the value to the to range
-    tmpValue *= scaleFactor;
-    // Re-zero back to the to range
-    return tmpValue + toLow;
+  static calcAQI(Cp: number, Ih: number, Il: number, BPh: number, BPl: number): number {
```

Comment:
Took me a min to understand c = concentration, I = AQI, and BP = breakpoint. Maybe just explain that in comments?